OctopusDSC/Tests/cTentacleAgent.Tests.ps1

```diff
@@ -255,353 +272,426 @@ try
+                    $params = Get-RequestedConfiguration "UninstallingRunningInstance"
+                }
                 Assert-ExpectedResult "UninstallingRunningInstance"
                 it "Should not download the MSI" {
+                    Set-TargetResource @params
                     Assert-MockCalled Request-File -times 0
                 }
                 it "Should not install the MSI" {
+                    Set-TargetResource @params
                     Assert-MockCalled Invoke-MsiExec -times 0
                 }
                 it "Should uninstall the MSI" {
+                    Set-TargetResource @params
                     Assert-MockCalled Invoke-MsiUninstall
                 }
                 it "Should not start the service" {
+                    Set-TargetResource @params
                     Assert-MockCalled Start-Service -times 0
                 }
             }
 
             Context "Uninstall running instance (with space)" {
-                Mock Invoke-TentacleCommand #{ write-host "`"$($args[1] -join ' ')`"," }
-                Mock Get-TargetResource { return Get-CurrentConfiguration "UninstallingRunningInstanceInSpace" }
-                Mock Invoke-MsiExec {}
-                Mock Invoke-MsiUninstall {}
-                Mock Request-File {}
-                Mock Update-InstallState {}
-                Mock Invoke-AndAssert {}
-                Mock Start-Service {}
-                Mock Get-CimInstance { return @() } # no other instances on the box
-                Mock Test-TentacleExecutableExists { return $true }
-
-                $params = Get-RequestedConfiguration "UninstallingRunningInstanceInSpace"
-                Set-TargetResource @params
-
+                BeforeAll {
+                    Mock Invoke-TentacleCommand #{ write-host "`"$($cmdArgs -join ' ')`"," }
+                    Mock Get-TargetResource { return Get-CurrentConfiguration "UninstallingRunningInstanceInSpace" }
+                    Mock Invoke-MsiExec {}
+                    Mock Invoke-MsiUninstall {}
+                    Mock Request-File {}
+                    Mock Update-InstallState {}
+                    Mock Invoke-AndAssert {}
+                    Mock Start-Service {}
+                    Mock Get-CimInstance { return @() } # no other instances on the box
+                    Mock Test-TentacleExecutableExists { return $true }
+                    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseDeclaredVarsMoreThanAssignments", "", Justification = "It is actually used, but pester's scoping is weird")]
+                    $params = Get-RequestedConfiguration "UninstallingRunningInstanceInSpace"
+                }
                 Assert-ExpectedResult "UninstallingRunningInstanceInSpace"
                 it "Should not download the MSI" {
+                    Set-TargetResource @params
```

Comment:
We used to call this in "setup" (ie, direct in the parent `context` block), but that is no longer in scope.
The only way i could get it to work is to call Set-TargetResource each time (rather than in the BeforeAll)